### PR TITLE
Support for Mesos resource-limits

### PR DESCRIPTION
**Pull request by timcharper**
JIRA Issues: D2IQ-62959, D2IQ-62961

### CODE REVIEW

**Comment by kaiwalyajoshi** [comment]
@timcharper :+1: 
Was thinking about adding more tests for the side-car task case where I think we'll get significant gains in terms of resource-reuse in our current payloads.

**Comment by timcharper** [comment]
Yeah, when Mesos support is ready it'd be great to have an actual integration test with Mesos to assert that the limits actually work

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/Capabilities.java [hunk 1]
```diff
@@ -130,6 +130,10 @@ public boolean supportsShm() {
     return hasOrExceedsVersion(1, 14);
   }
 
+  public boolean supportsResourceLimits() {
+    // Shared Mem fields supported 1.15 and above (Mesos 1.10)
+    return hasOrExceedsVersion(1, 15);
```
Comment (kaiwalyajoshi) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/Capabilities.java:135]:
```suggestion
    return hasOrExceedsVersion(2, 1);
```

File: docs/pages/developer-guide.md [hunk 2]
```diff
@@ -1732,6 +1732,9 @@ pods:
       hello-resources:
         cpus: 1.0
         memory: 256
+        resource-limits:
+          cpus: unlimited
+          mem: 512
```
Comment (janisz) [inline docs/pages/developer-guide.md:1737]:
```suggestion
          memory: 512
```

File: docs/pages/developer-guide.md [hunk 3]
```diff
@@ -1764,6 +1767,9 @@ pods:
         cmd: "echo hello >> shared-container-path/output && sleep 1000"
         cpus: 1.0
         memory: 256
+        resource-limits:
+          cpus: unlimited
+          mem: 512
```
Comment (janisz) [inline docs/pages/developer-guide.md:1772]:
```suggestion
          memory: 512
```

File: docs/pages/developer-guide.md (hunk 2 above)
Comment (janisz) [reply docs/pages/developer-guide.md:1737]: @kaiwalyajoshi @timcharper

**Review by kaiwalyajoshi — APPROVED** [review]

**Comment by kaiwalyajoshi** [comment]
This was merged in #3320 and scale tested in MWT-26.  SDK 0.58.0 rc1-4 has this feature.
Closing this PR.

**Pull request closed without merging**